cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ClusterCa.java

```diff
@@ -118,8 +117,7 @@ protected Map<String, CertAndKey> generateCcCerts(
             String clusterName,
             Map<String, CertAndKey> existingCertificates,
             Set<NodeRef> nodes,
-            boolean isMaintenanceTimeWindowsSatisfied,
-            boolean caCertGenerationChanged
```

Comment:
Can you please remind me why is this not needed anymore?

Reply:
For the Kafka certs we now have to check the generation id on multiple Secrets. So I've changed the logic so that instead of passing in the caCertGenerationChanged boolean here, we only pass through existing certificates that have the correct CA cert generation. You can see the matching changes in e.g. Cruise Control:

```
ccCerts = clusterCa.generateCcCerts(namespace, clusterName,
    // Only pass existing certificates if the CA cert generation hasn't changed since they were generated
    clusterCa.hasCaCertGenerationChanged(existingSecret) ? Map.of() : CertUtils.extractCertsAndKeysFromSecret(existingSecret, nodes),
    nodes, isMaintenanceTimeWindowsSatisfied);
```

Reply:
I did it this way so we wouldn't try to replace all certificates for the Kafka pods if only one of them was still out of date

Reply:
Right. Thanks for the clarification. That makes sense.

Reply:
@scholzj I'm not sure what you mean, I'm still passing through the `isMaintenanceTimeWindowsSatisfied` boolean

Reply:
You check the generation outside now. And when the generation changed, you remove it from the array. So inside the CA classes, this would be seen as a missing certificate and should be generated even when outside of the maintenance time window.

So the question is I guess whether it was like this even before and when the server certificates had old generation, were they recreated even outside of maintenance time windows?

Reply:
Yes I believe so, before if the cert generation changed or it is missing we immediately generate a new certificate, see the if statement on line 270 of this file:
```
if (!this.certRenewed() // No CA renewal is happening
        && certAndKey != null // There is a public cert and private key for this pod
        && !caCertGenerationChanged // The CA cert generation has not changed since the existing certificates were issued
    )   {
    ... various other checks
} else {
    LOGGER.debugCr(reconciliation, "Generating new certificate for node {}", node);
    CertAndKey k = generateSignedCert(subject, brokerCsrFile, brokerKeyFile, brokerCertFile, brokerKeyStoreFile);
    certs.put(podName, k);
}
```

I've just deleted the `!caCertGenerationChanged` check

Reply:
Ok, it should be good then. Thanks.